execution-engine/contracts/hdac-system/pop/src/pop_impl/mod.rs

```diff
@@ -228,20 +243,31 @@ impl ProofOfProfessionContract {
 
     // For user
     pub fn claim_reward(&self, user: &PublicKey) -> Result<()> {
-        let mut rewards = ContractClaim::read_reward()?;
-        let user_reward = rewards
+        let mut inflation_rewards = ContractClaim::read_reward(INFLATION_REWARD)?;
+        let user_inflation_reward = inflation_rewards
+            .0
+            .get(user)
+            .cloned()
+            .unwrap_or_revert_with(Error::RewardNotFound);
+        inflation_rewards.claim_rewards(user, &user_inflation_reward);
+        ContractClaim::write_reward(INFLATION_REWARD, &inflation_rewards);
+
+        let mut fare_rewards = ContractClaim::read_reward(FARE_REWARD)?;
```

Comment:
But recommend not to delete, just block them, and a developer far far later can be used.